### PR TITLE
Modifications for Cray build

### PR DESCRIPTION
**Pull request by mewall**
o Add CRAY_SDK macro
o Add "integer argc" to benchmarks, depending on CRAY_SDK 
o Disable testing and example builds in crusher script 
o Put prg_c_interface in new progress_c library
o Only build progress_c if not Cray compiler

- - -
This change is [Reviewable](https://reviewable.io/reviews/lanl/qmd-progress/283)

### CODE REVIEW

**Review by suemni — APPROVED** [review]

File: benchmarks/dmconstruction/dmconstruction.F90 [hunk 1]
```diff
@@ -28,6 +28,10 @@ program hmodel
   real(dp) :: ef,sparsity,dec,mlsi,mlsf,bnorm
   character(20) :: bml_dmode
 
+#ifdef CRAY_SDK
```
Comment (jeanlucf22) [inline benchmarks/dmconstruction/dmconstruction.F90:31]: Is it not correct to always define iargc as an integer?

File: CMakeLists.txt [hunk 2]
```diff
@@ -36,7 +36,7 @@ add_definitions(-DPROGRESS_VERSION=${PROGRESS_VERSION})
 message(STATUS "Setting PROGRESS_VERSION to '${PROGRESS_VERSION}'")
 
 if(CMAKE_Fortran_COMPILER_ID STREQUAL "Cray")
-  set(EXTRA_FCFLAGS "-ef ${EXTRA_FCFLAGS}")
+	set(EXTRA_FCFLAGS "-ef -DCRAY_SDK ${EXTRA_FCFLAGS}")
```
Comment (jeanlucf22) [inline CMakeLists.txt:39]: we should do that in build script, not here

**Comment by jeanlucf22** [comment]
Maybe a better way is to add a build option to build with or without C interface, instead of building two libraries?

**Comment by mewall** [comment]
Yes, I actually ran into problems with dependencies when making this change. I backed it out and included prg_c_interface conditional on not Cray.

I think the declaration of iargc as integer should be OK whatever the compiler. I'll remove the macro.
________________________________
From: Jean-Luc Fattebert ***@***.***>
Sent: Thursday, November 30, 2023 9:58 AM
To: lanl/qmd-progress ***@***.***>
Cc: Wall, Michael E ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [lanl/qmd-progress] Modifications for Cray build (PR #283)


Maybe a better way is to add a build option to build with or without C interface, instead of building two libraries?

**Review by jeanlucf22 — APPROVED** [review]

**Pull request merged**